### ISSUE TITLE
`az login` fails with `UnboundLocalError: local variable 'token_entry' referenced before assignment`

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#15961]
**Describe the bug**

#15776 introduced a bug: `az login` fails with `UnboundLocalError: local variable 'token_entry' referenced before assignment`

When ADAL login fails, the actual error is hidden by `_login_exception_handler`.

https://github.com/Azure/azure-cli/blob/3ade968815d0d70556981b3fa23809f54aed1979/src/azure-cli-core/azure/cli/core/_profile.py#L1373-L1380

To repro, login with an invalid password:

```powershell
> az login -u "rolelivetest@azuresdkteam.onmicrosoft.com" -p "xxx"
CLIInternalError: The command failed with an unexpected error. Here is the traceback:
local variable 'token_entry' referenced before assignment
Traceback (most recent call last):
  File "d:\cli\knack\knack\cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 654, in execute
    raise ex
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 717, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 710, in _run_job
    six.reraise(*sys.exc_info())
  File "D:\cli\env38\lib\site-packages\six.py", line 703, in reraise
    raise value
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\__init__.py", line 784, in default_command_handler
    return op(**command_args)
  File "d:\cli\azure-cli\src\azure-cli\azure\cli\command_modules\profile\custom.py", line 150, in login
    subscriptions = profile.find_subscriptions_on_login(
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\_profile.py", line 212, in find_subscriptions_on_login
    subscriptions = subscription_finder.find_from_user_account(
  File "d:\cli\azure-cli\src\azure-cli-core\azure\cli\core\_profile.py", line 851, in find_from_user_account
    if not token_entry:
UnboundLocalError: local variable 'token_entry' referenced before assignment
To open an issue, please run: 'az feedback'
```

**Comment by yonzhan** [comment]
az login

**Comment by jiasli** [comment]
This issue usually happens when the 

- the user credential is invalid, or 
- the HTTP proxy is invalid.

To fix it, please 

- use a valid user credential, or 
- confirm the HTTP proxy setting.

We are releasing a hotfix to expose the real error message. Sorry for the inconvenience caused and thanks for your kind understanding.

**Comment by jiasli** [comment]
We have released **2.15.1** to fix this issue. Please [install the latest version](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) and kindly let us know if the issue is resolved.
